### PR TITLE
Action refactoring - split the run-in-renode.py

### PR DESCRIPTION
**Pull request by WiktorOgrodnik**
The run-in-renode.py script was splitted into smaller files to improve code readability and facilitate later changes.

-action arguments are now converted to json
 before being passed to the script
-small typos fixes
-replaces sendcontrol('c') with sendline('') in some places -fixes potential races with git repos sideloading
-adds 1 hour timeout to installing pip packages on target

### CODE REVIEW

File: action/run-in-renode.py [hunk 1]
```diff
@@ -610,23 +247,34 @@ def run_cmds_in_renode(commands_to_run: str):
...
+        # after required Python version will increases to >= 3.10
+
+        if arg == "renode-run":
+            renode_run = args[arg]
+        elif arg == "devices":
+            add_devices(args[arg])
+        elif arg == "python-packages":
+            add_packages(args[arg])
+        elif arg == "repos":
+            add_repos(args[arg])
```
Comment (kbieganski) [inline action/run-in-renode.py:271]:
```suggestion
    renode_run = args.get('renode-run', default=None)
    add_devices(args.get('devices', default=''))
    add_packages(args.get('python-packages', default='')
    add_repos(args.get('repos', default=''))
```
Or sth like that?

File: action/run-in-renode.py [hunk 2]
```diff
@@ -530,6 +165,7 @@ def setup_renode():
         run_cmd(child, "#", 'echo "nameserver 1.1.1.1" >> /etc/resolv.conf')  # adds dns server address
 
         child.expect_exact("#")
+        child.logfile_read = None
```
Comment (kbieganski) [inline action/run-in-renode.py:168]: Add a comment why

File: action/run-in-renode.py [hunk 3]
```diff
@@ -540,11 +176,11 @@ def setup_python():
     Install previously downloaded python packages in emulated linux.
     """
 
-    child = px_spawn("telnet 127.0.0.1 1234", encoding="utf-8", timeout=None)
+    child = px_spawn("telnet 127.0.0.1 1234", encoding="utf-8", timeout=10)
```
Comment (kbieganski) [inline action/run-in-renode.py:179]: Why add the timeout?

File: action/run-in-renode.py [hunk 4]
```diff
@@ -540,11 +176,11 @@ def setup_python():
...
 
     try:
-        child.expect_exact("'^]'.")
-        child.sendcontrol("c")
+        child.expect_exact("'^]'.", timeout=30)
+        child.sendline('')
```
Comment (kbieganski) [inline action/run-in-renode.py:183]: Why this change?

**Review by kbieganski — CHANGES_REQUESTED** [review]
Any significant changes in the stuff refactored to other files?

File: action/run-in-renode.py (hunk 4 above)
Comment (WiktorOgrodnik) [reply action/run-in-renode.py:183]: I have added timeouts to every command, so i think there will not be such situations like 6h random tests.

File: action/run-in-renode.py (hunk 2 above)
Comment (WiktorOgrodnik) [reply action/run-in-renode.py:168]: I think it is not necessary, I can delete that

File: action/run-in-renode.py (hunk 1 above)
Comment (WiktorOgrodnik) [reply action/run-in-renode.py:271]: Looks great

File: action/run-in-renode.py (hunk 4 above)
Comment (WiktorOgrodnik) [reply action/run-in-renode.py:183]: I replaced sendcontrol('c') with sendline('') because as I see this second approach is more idiomatic. And I think it is kind of rude to send interrupts when they are not needed :)

**Comment by WiktorOgrodnik** [comment]
@kbieganski Unfortunately, this github commit diff is not very helpful, because it does not show code moves, and these new files look like a big code addition, but the rest of the changes are minor.

**Review by kbieganski — APPROVED** [review]
